diff -ru OpenGL-Simple-0.03/Makefile.PL OpenGL-Simple-0.04/Makefile.PL
--- OpenGL-Simple-0.03/Makefile.PL	2005-11-25 04:24:48.000000000 +0100
+++ OpenGL-Simple-0.04/Makefile.PL	2009-08-12 23:43:21.498609863 +0200
@@ -72,6 +72,7 @@
 
 		GL_TEXTURE_2D GL_TEXTURE_GEN_Q GL_TEXTURE_GEN_R
 		GL_TEXTURE_GEN_S GL_TEXTURE_GEN_T
+		GL_TEXTURE_BINDING_1D GL_TEXTURE_BINDING_2D
 
         GL_Q GL_R GL_S GL_T
 
diff -ru OpenGL-Simple-0.03/Simple.pm OpenGL-Simple-0.04/Simple.pm
--- OpenGL-Simple-0.03/Simple.pm	2005-11-25 04:24:48.000000000 +0100
+++ OpenGL-Simple-0.04/Simple.pm	2009-08-12 23:45:55.696049999 +0200
@@ -123,6 +123,7 @@
 		GL_SCISSOR_TEST GL_STENCIL_TEST GL_TEXTURE_1D
 		GL_TEXTURE_2D GL_TEXTURE_GEN_Q GL_TEXTURE_GEN_R
 		GL_TEXTURE_GEN_S GL_TEXTURE_GEN_T
+		GL_TEXTURE_BINDING_1D GL_TEXTURE_BINDING_2D
 
                 GL_Q GL_R GL_S GL_T
 
@@ -357,7 +358,7 @@
 
 	# Preserve old 9-anonymous-argument version.
 	if (9==@_) {
-		realGLTexImage2D(@_);
+		realglTexImage2D(@_);
 		return;
 	}
 
diff -ru OpenGL-Simple-0.03/Simple.xs OpenGL-Simple-0.04/Simple.xs
--- OpenGL-Simple-0.03/Simple.xs	2005-11-25 04:45:14.000000000 +0100
+++ OpenGL-Simple-0.04/Simple.xs	2009-08-12 23:34:06.559787461 +0200
@@ -751,9 +751,9 @@
 			 * These are defined as GL_TEXTURE_nD_BINDING_EXT
 			 * in my nvidia headers. Need to work it out
 			 * on the fly, I guess.
-			case GL_TEXTURE_1D_BINDING:
-			case GL_TEXTURE_2D_BINDING:
 			*/
+			case GL_TEXTURE_BINDING_1D:
+			case GL_TEXTURE_BINDING_2D:
 			case GL_TEXTURE_COORD_ARRAY_SIZE:
 			case GL_TEXTURE_COORD_ARRAY_STRIDE:
 			case GL_TEXTURE_COORD_ARRAY_TYPE:
